src/mbgl/map/transform.cpp

```diff
@@ -243,8 +243,8 @@ void Transform::flyTo(const CameraOptions &camera, const AnimationOptions &anima
 
     // When u₀ = u₁, the optimal path doesn’t require both ascent and descent.
     bool isClose = std::abs(u1) < 0.000001;
-    // Perform a more or less instantaneous transition if the path is too short.
-    if (isClose && std::abs(w0 - w1) < 0.000001) {
+    // Perform a more or less instantaneous transition if the path is too short or zoom in/out factor is infinte
+    if ((isClose && std::abs(w0 - w1) < 0.000001) || std::isinf(r(0)) || std::isinf(r(1))) {
```

Comment:
Been looking back and forth between the [original paper](https://www.win.tue.nl/~vanwijk/zoompan.pdf) and our implementation. One difference that I noticed is that we added a check for small changes, as these aren't ideal for the algorithm. Instead we are currently performing an easeTo instead. I don't feel that I'm adding a band-aid to the problem, I'm hardening the existing one.

<img width="313" alt="screen shot 2017-03-13 at 14 31 36" src="https://cloud.githubusercontent.com/assets/2151639/23856427/e67569fa-07f9-11e7-9c64-53585f74644e.png">

Above is a normal situation of a flyTo implementation but the use-case described in
https://github.com/mapbox/mapbox-gl-native/issues/8300 is having a origin and target camera position that is almost the same. This isn't fully captured by current bandaid solution. The algorithm seem to generate -infinite zoom out values as a result of log(o) in following code with i=1:

```cpp
/** rᵢ: Returns the zoom-out factor at one end of the animation.
  @param i 0 for the ascent or 1 for the descent. */
auto r = [=](double i) {
            /// bᵢ
      double b = (w1 * w1 - w0 * w0 + -1 * rho2 * rho2 * u1 * u1) / (2 * (i ? w1 : w0) * rho2 * u1);
      return std::log(std::sqrt(b * b + 1) - b);
};
```

My proposal is to optimise the bandaid solution as this interaction is not ideal for a flyTo interaction.